### PR TITLE
Update Bin entity creation API

### PR DESCRIPTION
**Pull request by pablogs9**
Related:
https://github.com/micro-ROS/rmw_microxrcedds/pull/213
https://github.com/eProsima/Micro-XRCE-DDS/pull/118

### CODE REVIEW

**Comment by richiprosima** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS-Client%20Manual%20Linux/506/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS-Client%20Manual%20Windows/522/)

File: examples/BinaryEntityCreation/main.c [hunk 1]
```diff
@@ -95,16 +95,20 @@ int main(
                     UXR_REPLACE);
 
     uxrObjectId datawriter_id = uxr_object_id(0x01, UXR_DATAWRITER_ID);
+    uxrQoS_t qos = {
+        .reliability = UXR_RELIABILITY_RELIABLE, .durability = UXR_DURABILITY_TRANSIENT_LOCAL,
+        .history = UXR_HISTORY_KEEP_LAST, .depth = 1
```
Comment (Acuadros95) [inline examples/BinaryEntityCreation/main.c:100]:
```suggestion
        .history = UXR_HISTORY_KEEP_LAST, .depth = 0
```

In this example, `depth = 0`

File: examples/PublishHelloWorldCAN/main.c [hunk 2]
```diff
@@ -80,8 +80,12 @@ int main(
                     UXR_REPLACE);
 
     uxrObjectId datawriter_id = uxr_object_id(0x01, UXR_DATAWRITER_ID);
+    uxrQoS_t qos = {
+        .reliability = UXR_RELIABILITY_RELIABLE, .durability = UXR_DURABILITY_TRANSIENT_LOCAL,
+        .history = UXR_HISTORY_KEEP_LAST, .depth = 1
```
Comment (Acuadros95) [inline examples/PublishHelloWorldCAN/main.c:85]:
```suggestion
        .history = UXR_HISTORY_KEEP_LAST, .depth = 0
```

File: test/shared_memory/SharedMemory.cpp [hunk 3]
```diff
@@ -57,6 +57,8 @@ class SharedMemoryTest : public ::testing::TestWithParam<XRCECreationMode>
 
 protected:
 
+    uxrQoS_t qos = {UXR_DURABILITY_TRANSIENT_LOCAL, UXR_RELIABILITY_RELIABLE, UXR_HISTORY_KEEP_LAST, 1};
```
Comment (Acuadros95) [inline test/shared_memory/SharedMemory.cpp:60]:
```suggestion
    uxrQoS_t qos = {UXR_DURABILITY_TRANSIENT_LOCAL, UXR_RELIABILITY_RELIABLE, UXR_HISTORY_KEEP_LAST, 0};
```

File: include/uxr/client/core/session/create_entities_bin.h [hunk 4]
```diff
@@ -139,10 +161,7 @@ typedef enum uxrQoSDurability
  * @param object_id             The identifier of the XRCE DataWriter.
  * @param publisher_id          The identifier of the associated XRCE Publisher.
  * @param topic_id              The identifier of the associated XRCE Topic.
- * @param reliable              Reliability flag.
- * @param keep_last             Keep last flag.
- * @param depth                 Depth of the history.
- * @param transient_local       Transient local flag.
+ * @param qos              QoS definition.
```
Comment (Acuadros95) [inline include/uxr/client/core/session/create_entities_bin.h:164]:
```suggestion
 * @param qos                   QoS definition.
```

**Comment by richiprosima** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS-Client%20Manual%20Linux/507/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/Micro-XRCE-DDS-Client%20Manual%20Windows/523/)

**Review by Acuadros95 — APPROVED** [review]

**Pull request merged**